src/react/utils.js

```diff
@@ -797,36 +797,28 @@ export function getValueFromFunctionCall(
   invariant(func.$Call, "Expected function to be a FunctionValue with $Call method");
   let funcCall = func.$Call;
   let newCall = func.$Construct;
-  let effects;
+  let completion;
   try {
-    effects = realm.evaluateForEffects(
-      () => {
-        invariant(func);
-        if (isConstructor) {
-          invariant(newCall);
-          return newCall(args, func);
-        } else {
-          return funcCall(funcThis, args);
-        }
-      },
-      null,
-      "getValueFromFunctionCall"
-    );
+    if (isConstructor) {
+      invariant(newCall);
+      completion = newCall(args, func);
+    } else {
+      completion = funcCall(funcThis, args);
+    }
   } catch (error) {
-    throw error;
+    if (error instanceof AbruptCompletion) {
+      completion = error;
+    } else {
+      throw error;
+    }
   }
-
-  let completion = effects.result;
   if (completion instanceof PossiblyNormalCompletion) {
     // in this case one of the branches may complete abruptly, which means that
     // not all control flow branches join into one flow at this point.
     // Consequently we have to continue tracking changes until the point where
     // all the branches come together into one.
     completion = realm.composeWithSavedCompletion(completion);
```

Comment:
Doing this while the effects that contains completion have not been applied causes completion to capture the current outer generator, which is why the declaration is out of place later on.